renku_notebooks/util/kubernetes_.py

```diff
@@ -56,11 +57,30 @@ def secret_exists(name, k8s_client, k8s_namespace):
     return False
 
 
-def make_server_name(safe_username, namespace, project, branch, commit_sha):
-    """Form a 16-digit hash server ID."""
-    server_string = f"{safe_username}-{namespace}-{project}-{branch}-{commit_sha}"
-    return "{username}-{project}-{hash}".format(
-        username=safe_username[:10].lower(),
+def make_server_name(
+    safe_username: str, namespace: str, project: str, branch: str, commit_sha: str
+) -> str:
+    """Form a unique server name.
+
+    This is used in naming all of the k8s resources created by amalthea.
+    """
+    server_string_for_hashing = (
+        f"{safe_username}-{namespace}-{project}-{branch}-{commit_sha}"
+    )
+    safe_username_lowercase = safe_username.lower()
+    if re.match(r"[a-z]", safe_username_lowercase[0]):
```

Comment:
```suggestion
    if safe_username_lowercase[0].isalpha() and safe_username_lowercase[0].isascii():
```
No need for regex :slightly_smiling_face: Also then you can just lowercase on the `username=...` line since this works for both upper/lower case.

You could also just set prefix and overwrite in in an `if` then no `else` is needed.